src/classes/adapter.js

```diff
@@ -0,0 +1,169 @@
+const Base = require('./base')
+const Knex = require('knex')
+const {get} = require('lodash')
+
+module.exports = class Adapter extends Base {
+  /**
+   * Sets up the knex query builder instance
+   * @constructor
+   */
+  constructor (store, options) {
+    super()
+    this.store = store
+    this.knex = Knex({
+      client: 'pg',
+      connection: options,
+      searchPath: 'knex,public'
+    })
+  }
+
+  fieldsForModel (Model) {
+    return Object.keys(get(Model, 'definition.attributes'), {})
+  }
+
+  include (Model, query, includeString = '', fieldOptions) {
+    return query.then(data => {
+      data = JSON.parse(JSON.stringify(data))
+      if (!Array.isArray(data)) data = [data]
+      return Promise.all(includeString.split(',').map(include => {
+        include = include.trim()
+        if (include.indexOf('.') !== -1) return
+        const definition = get(Model, `definition.relationships.${include}`)
+        if (!definition) return
+        const RelatedModel = this.store.modelFor(definition.modelTo)
+        if (!RelatedModel) return
+        let query = this.knex(RelatedModel.tableName)
+
+        let attributes = this.fieldsForModel(RelatedModel)
+        if (fieldOptions) attributes = this.limitFieldSet(RelatedModel, attributes, fieldOptions)
+        query = query.column(attributes)
+
+        let ids = data.map(item => item[definition.keyFrom])
+        query = query.whereIn(definition.keyTo, ids)
+
+        return query.then(relatedData => {
+          const relatedDataGroups = new Map()
+          relatedData.forEach(relData => {
+            if (relatedDataGroups.has(relData[definition.keyTo])) {
+              relatedDataGroups.get(relData[definition.keyTo]).push(relData)
+            } else {
+              relatedDataGroups.set(relData[definition.keyTo], [relData])
+            }
+          })
+          return {name: include, definition, groupedData: relatedDataGroups}
+        })
+      }))
+      .then(relationshipData => {
+        data.forEach(item => {
+          relationshipData.forEach(relationship => {
+            const relData = relationship.groupedData.get(item[relationship.definition.keyFrom])
+            if (relationship.definition.type === 'belongsTo') {
+              item[relationship.name] = relData[0]
+            } else {
+              item[relationship.name] = relData
+            }
+          })
+        })
+        return JSON.parse(JSON.stringify(data))
+      })
+    })
+  }
```

Comment:
There it is... round 1. Passes some basic tests and works for 1 level of relationship including. Good luck reading that. I know I'm not looking forward to trying to understand what I just wrote when it comes to round 2.